src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -3563,7 +3547,17 @@ public override Type[] GetGenericParameterConstraints()
         #endregion
 
         #region Misc
-        internal bool IsNullableOfT => Cache.IsNullableOfT;
+        internal unsafe bool IsNullableOfT
+        {
+            get
+            {
+                TypeHandle th = GetNativeTypeHandle();
+
+                bool isNullable = !th.IsTypeDesc && th.AsMethodTable()->IsNullable;
```

Comment:
Just confirm: based on original behavior of `IsNullableOfT`, it will return false when type is` Nullable<>` definition, but `IsNullable` property is to check method table flags (0x00050000 with mask 0x000F0000) and `Nullable<>` definition type also sets this flag, so not sure if need to have additional check here.

Reply:
Good catch! Yes, this is intentional simplification. IsNullableOfT property is only ever used on constructed generic types. It does not matter what it returns for Nullable type definition.